code/Model/UserDefinedForm.php

```diff
@@ -0,0 +1,405 @@
+            $fields->addFieldToTab('Root.Recipients', $emailRecipients);
+            $fields->addFieldsToTab('Root.FormOptions', $this->getFormOptions());
+
+
+            // view the submissions
+            // make sure a numeric not a empty string is checked against this int column for SQL server
+            $parentID = (!empty($this->ID)) ? (int) $this->ID : 0;
+
+            // get a list of all field names and values used for print and export CSV views of the GridField below.
+            $columnSQL = <<<SQL
+SELECT "SubmittedFormField"."Name" as "Name", COALESCE("EditableFormField"."Title", "SubmittedFormField"."Title") as "Title", COALESCE("EditableFormField"."Sort", 999) AS "Sort"
+FROM "SubmittedFormField"
+LEFT JOIN "SubmittedForm" ON "SubmittedForm"."ID" = "SubmittedFormField"."ParentID"
+LEFT JOIN "EditableFormField" ON "EditableFormField"."Name" = "SubmittedFormField"."Name" AND "EditableFormField"."ParentID" = '$parentID'
+WHERE "SubmittedForm"."ParentID" = '$parentID'
+ORDER BY "Sort", "Title"
+SQL;
+            // Sanitise periods in title
+            $columns = array();
+            foreach (DB::query($columnSQL)->map() as $name => $title) {
+                $columns[$name] = trim(strtr($title, '.', ' '));
+            }
+
+            $config = GridFieldConfig::create();
+            $config->addComponent(new GridFieldToolbarHeader());
+            $config->addComponent($sort = new GridFieldSortableHeader());
+            $config->addComponent($filter = new UserFormsGridFieldFilterHeader());
+            $config->addComponent(new GridFieldDataColumns());
+            $config->addComponent(new GridFieldEditButton());
+            $config->addComponent(new GridFieldDeleteAction());
+            $config->addComponent(new GridFieldPageCount('toolbar-header-right'));
+            $config->addComponent($pagination = new GridFieldPaginator(25));
+            $config->addComponent(new GridFieldDetailForm());
+            $config->addComponent(new GridFieldButtonRow('after'));
+            $config->addComponent($export = new GridFieldExportButton('buttons-after-left'));
+            $config->addComponent($print = new GridFieldPrintButton('buttons-after-left'));
+
+            // show user form items in the summary tab
+            $summaryarray = array(
+                'ID' => 'ID',
+                'Created' => 'Created',
+                'LastEdited' => 'Last Edited'
+            );
+            foreach (EditableFormField::get()->filter(array('ParentID' => $parentID)) as $eff) {
+                if ($eff->ShowInSummary) {
+                    $summaryarray[$eff->Name] = $eff->Title ?: $eff->Name;
+                }
+            }
+
+            $config->getComponentByType(GridFieldDataColumns::class)->setDisplayFields($summaryarray);
+
+            /**
+             * Support for {@link https://github.com/colymba/GridFieldBulkEditingTools}
+             */
+            if (class_exists(BulkManager::class)) {
```

Comment:
We might need a FQCN here.

Reply:
`::class` uses the imported class name (includes namespace)